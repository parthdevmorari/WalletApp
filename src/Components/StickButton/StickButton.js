import React from 'react';
import { TouchableOpacity, View } from 'react-native';
import * as resources from 'resources';
import styles from './StickButtonStyle';
import Icon from "react-native-vector-icons/Feather";

const StickButton = ({ value, onPress, style, iconStyle, size = 22, ...props }) => (
	<TouchableOpacity style={[ styles.plusButton, style ]} onPress={onPress} {...props}>
		<View style={styles.plusButtonView}>
			<Icon name="plus" style={[styles.iconStyle,iconStyle]} />
		</View>
	</TouchableOpacity>
);

export default StickButton;
