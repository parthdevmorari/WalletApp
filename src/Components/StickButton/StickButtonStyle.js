import { StyleSheet, Dimensions, StatusBar } from 'react-native';
import { widthPercentageToDP as wp, heightPercentageToDP as hp } from 'react-native-responsive-screen';
import { color } from "../../Constants/Color";

const width = Dimensions.get('window').width;
const ITEM_WIDTH = Math.round(width - 70);
const styles = StyleSheet.create({
	plusButton: {
		position:'absolute',
		right:15,
		bottom:15,
		// zIndex:9,
		width:45,
		height:45,
		backgroundColor: color._0075bf,
		borderRadius: 200,
		justifyContent:'center',
		alignItems:'center',
	},
	plusButtonView:{
		shadowColor: "#000",
		shadowOffset: {
			width: 0,
			height: 4,
		},
		shadowOpacity: 0.30,
		shadowRadius: 4.65,

		elevation: 8,
	},
	iconStyle: {
		fontSize: wp(10),
		color:'#fff',
	}

});
export default styles;

