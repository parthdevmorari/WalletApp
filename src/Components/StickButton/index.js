import component from './StickButton';
export default component;