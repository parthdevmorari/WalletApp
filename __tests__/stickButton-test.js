import 'react-native';
import React from 'react';
import StickButton from '../src/Components/StickButton/StickButton'
import Enzyme, {shallow, mount} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJson from 'enzyme-to-json';

Enzyme.configure({adapter: new Adapter()});

describe('StickButton Component', () => {
  test('StickButton Component', () => {
    const wrapper = shallow(<StickButton />);
    expect(wrapper.exists()).toBe(true);
  });
  it('StickButton  toMatchSnapshot', () => {
    const wrapper = shallow(<StickButton />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});