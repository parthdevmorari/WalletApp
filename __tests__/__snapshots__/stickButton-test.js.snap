// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StickButton Component StickButton  toMatchSnapshot 1`] = `
<TouchableOpacity
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#0075bf",
        "borderRadius": 375,
        "bottom": 15,
        "height": 84.5,
        "justifyContent": "center",
        "position": "absolute",
        "right": 15,
        "width": 84.5,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "elevation": 8,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 4,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 4.65,
      }
    }
  >
    <Icon
      allowFontScaling={false}
      name="plus"
      size={12}
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 75,
          },
          undefined,
        ]
      }
    />
  </View>
</TouchableOpacity>
`;
